### PR TITLE
[AMSDK-10955] AEP Edge network Documentation for Analytics Edge

### PR DESCRIPTION
**Pull request by rymorale**

### CODE REVIEW

**Comment by rymorale** [comment]
WIP, opening the pr to get some feedback. please let me know if anything should be changed / improved. thanks!

File: using-mobile-extensions/adobe-analytics-edge/adobe-analytics-edge-release-notes.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Release Notes
+
+## January XX, 2021 (TO BE UPDATED)
```
Comment (emdobrin) [inline using-mobile-extensions/adobe-analytics-edge/adobe-analytics-edge-release-notes.md:3]: In my opinion we can either keep this page empty until the 1.0.0 release or add a note about beta for now.

File: using-mobile-extensions/adobe-analytics-edge/analytics-edge-api-reference.md [hunk 2]
```diff
@@ -0,0 +1,1249 @@
+# Analytics Edge API reference
+
+{% hint style="info" %}
+
+The following Analytics API are currently **not** supported with Analytics Edge. They are listed for reference only.
```
Comment (emdobrin) [inline using-mobile-extensions/adobe-analytics-edge/analytics-edge-api-reference.md:5]: We should move trackAction and trackState APIs on top and create a table/list with all the APIs which are not supported below (just API name, no samples). Also, I think we can remove the sample code for the other platforms other than ios/android since it is not supported.

**Review by emdobrin — CHANGES_REQUESTED** [review]

File: SUMMARY.md [hunk 3]
```diff
@@ -38,11 +38,21 @@
...
   * [Platform Services](using-mobile-extensions/mobile-core/platform-services.md)
-* [Adobe Analytics](using-mobile-extensions/adobe-analytics/README.md)
+  
+  [Adobe Analytics](using-mobile-extensions/adobe-analytics/README.md)
+  
   * [Release Notes](using-mobile-extensions/adobe-analytics/adobe-analytics-release-notes.md)
   * [Analytics API reference](using-mobile-extensions/adobe-analytics/analytics-api-reference.md)
   * [Products variable](using-mobile-extensions/adobe-analytics/product-variable.md)
+  
+* [Adobe Analytics Edge](using-mobile-extensions/adobe-analytics-edge/README.md)
```
Comment (emdobrin) [inline SUMMARY.md:48]: Adobe Analytics Edge Beta

File: using-mobile-extensions/adobe-analytics-edge/analytics-edge-api-reference.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
+# Analytics Edge API reference
```
Comment (emdobrin) [inline using-mobile-extensions/adobe-analytics-edge/analytics-edge-api-reference.md:1]: We should add this:
```
{% hint style="warning" %}

The AEP Analytics Edge mobile extension is currently in BETA. Use of this extension is by invitation only. Please contact your Adobe Customer Success Manager to learn more.

{% endhint %}
```

File: using-mobile-extensions/adobe-analytics-edge/product-variable.md [hunk 5]
```diff
@@ -84,67 +84,9 @@ contextData["m.purchase"] = "1"
...
-
-// send the tracking call - use either a trackAction or TrackState call.
-// trackAction example:
-FlutterACPCore.trackAction("purchase", data: contextData);
-// trackState example:
-FlutterACPCore.trackState("Order Confirmation", data: contextData);
+MobileCore.track(state: "Order Confirmation", data: contextData)
 ```
 {% endtab %}
 {% endtabs %}
```
Comment (emdobrin) [inline using-mobile-extensions/adobe-analytics-edge/product-variable.md:92]: We should probably remove the sample network request below and just add the description in text👇

File: using-mobile-extensions/adobe-analytics-edge/product-variable.md [hunk 6]
```diff
@@ -226,53 +168,9 @@ contextData["m.purchase"] = "1"
...
-// send the tracking call - use either a trackAction or TrackState call.
-// trackAction example:
-FlutterACPCore.trackAction("purchase", data: contextData);
+MobileCore.track(action: "purchase", data: contextData)
 // trackState example:
-FlutterACPCore.trackState("Order Confirmation", data: contextData);
+MobileCore.track(state: "Order Confirmation", data: contextData)
 ```
 {% endtab %}
 {% endtabs %}
```
Comment (emdobrin) [inline using-mobile-extensions/adobe-analytics-edge/product-variable.md:176]: @imironchuk do we still want to keep this example in here? I would prefer including an example with XDM products instead.

File: using-mobile-extensions/adobe-analytics-edge/README.md [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+
+   {% endtabs %}
+
+### Register Analytics Edge with Mobile Core
+
+{% tabs %}
+{% tab title="Android" %}
+#### Java
+
+To call the set up methods that call the [setApplication\(\)](https://aep-sdks.gitbook.io/docs/using-mobile-extensions/mobile-core/mobile-core-api-reference#setapplication) method in the `onCreate()` method:
```
Comment (emdobrin) [inline using-mobile-extensions/adobe-analytics-edge/README.md:81]: Let's rephrase this to something like:
In your Application class, register Analytics Edge with Mobile Core in the `onCreate()` method.

File: using-mobile-extensions/adobe-analytics-edge/analytics-edge-api-reference.md [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+let additionalContextData = ["key":"value"]
+
+// send the trackState call.
+MobileCore.track(state: "State Name", data: additionalContextData)
+```
+
+{% endtab %}
+{% endtabs %}
+
+## Unsupported API
```
Comment (emdobrin) [inline using-mobile-extensions/adobe-analytics-edge/analytics-edge-api-reference.md:111]: Can we add something like `If you are migrating from the Analytics to Analytics Edge extension, the following API are are no longer supported:`

and maybe have a table structure with 3 rows Analytics Queue APIs , AID API, VID APIs and add a few details on each of them.
```
listed Analytics Queue APIs -> the events batching and queuing is no longer supported in Analytics Edge, the hits are sent as soon as a network connection is available on the device
AID API -> if you used AID in the past, the SDK will continue to read it and send it to Experience Edge; for new users the recommended unique identifier is ECID(link to identity).
listed VID APIs -> if you used VID in your previous implementations,  the SDK will continue to read it and send it to Experience Edge; for new users the recommended unique identifier is the ECID(link to identity).

File: using-mobile-extensions/adobe-analytics-edge/product-variable.md [hunk 9]
```diff
@@ -57,9 +57,9 @@ NSMutableDictionary *contextData = [NSMutableDictionary dictionary];
```
Comment (emdobrin) [inline using-mobile-extensions/adobe-analytics-edge/product-variable.md:57]: The  Implementing a Merchandising Variable link no longer works, I think this is the new link https://experienceleague.adobe.com/docs/analytics/implementation/vars/plugins/addproductevar.html

**Review by emdobrin — APPROVED** [review]

File: using-mobile-extensions/adobe-analytics-edge/README.md [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+
+The AEP Analytics Edge mobile extension is currently in BETA. Use of this extension is by invitation only. Please contact your Adobe Customer Success Manager to learn more.
+
+{% endhint %}
+
+## **Configure the Adobe Experience Platform Edge extension in** Experience Platform **Launch**
+
+1. In Experience Platform Launch, click the **Extensions** tab.
+2. On the **Catalog** tab, locate the **Adobe Experience Platform Edge** extension, and click **Install**.
+3. Within the Edge Extension configuration page add your Edge configuration and environment settings. For more information on creating an Edge configuration for use with your mobile app, see [Experience Platform Setup](../../beta/experience-platform-extension/experience-platform-setup). Additionally, Analytics Edge extension setup steps are available in a tutorial at [Enable Analytics in Edge configuration.](../../beta/experience-platform-extension/tutorials/tutorial-5-analytics-edge-extension#enable-analytics-in-edge-configuration)
```
Comment (shalehaha) [inline using-mobile-extensions/adobe-analytics-edge/README.md:13]: should the links point to the new Edge extension docs?

File: using-mobile-extensions/adobe-analytics-edge/analytics-edge-api-reference.md [hunk 11]
```diff
@@ -0,0 +1,126 @@
+# Analytics Edge API reference
+
+{% hint style="warning" %}
+
+The AEP Analytics Edge mobile extension is currently in BETA. Use of this extension is by invitation only. Please contact your Adobe Customer Success Manager to learn more.
+
+{% endhint %}
+
+## trackAction
```
Comment (shalehaha) [inline using-mobile-extensions/adobe-analytics-edge/analytics-edge-api-reference.md:9]: do we still need those APIs to be documented here since they have already been documented in Core?

File: using-mobile-extensions/adobe-analytics-edge/README.md [hunk 12]
```diff
@@ -0,0 +1,199 @@
...
+{% tab title="iOS" %}
+1. Add the Mobile Core, Edge, Identity, and Analytics Edge extensions to your project using Cocoapods.
+
+2. Add the following pods in your `Podfile`:
+
+   ```ruby
+    pod 'AEPCore'
+    pod 'AEPEdge'
+    pod 'AEPIdentity'
+    pod 'AEPAnalyticsEdge', :git => 'https://github.com/adobe/aepsdk-analyticsedge-ios', :branch => 'main'
```
Comment (praveek) [inline using-mobile-extensions/adobe-analytics-edge/README.md:47]: pod 'AEPAnalyticsEdge', :git => 'https://github.com/adobe/aepsdk-analyticsedge-ios', :tag => '1.0.0-beta.1'

**Review by praveek — APPROVED** [review]

**Comment by praveek** [comment]
@imironchuk Should we add any info/warning about legacy Analytics extension or migration path ?

File: using-mobile-extensions/adobe-analytics-edge/README.md (hunk 10 above)
Comment (rymorale) [reply using-mobile-extensions/adobe-analytics-edge/README.md:13]: i'll update this

File: using-mobile-extensions/adobe-analytics-edge/analytics-edge-api-reference.md (hunk 11 above)
Comment (rymorale) [reply using-mobile-extensions/adobe-analytics-edge/analytics-edge-api-reference.md:9]: i think we can remove the track API from this page and just link to the Core docs. I'll leave the information about the unsupported API.

File: using-mobile-extensions/adobe-analytics-edge/product-variable.md (hunk 6 above)
Comment (emdobrin) [reply using-mobile-extensions/adobe-analytics-edge/product-variable.md:169]: @rymorale @praveek  We discussed about this and we decided to have two examples: first one using xdm product and the second one using &&product as described in this doc.

Comment (rymorale) [reply using-mobile-extensions/adobe-analytics-edge/product-variable.md:169]: i'll work on adding the xdm product example
